src/boot/efi/smbios.c

```diff
@@ -181,3 +201,44 @@ const char* smbios_find_oem_string(const char *name) {
 
         return NULL;
 }
+
+static const char* smbios_get_string(const SmbiosHeader *header, size_t nr, size_t left) {
+        assert(header);
+        const char *s = (const char *) header;
+
+        assert(left >= header->length);
```

Comment:
as discussed below, I would make these things graceful, i.e. validate things here and return NULL if this doesn't work out. don't trust shitty firmwares

Reply:
so, as discussed in the other PR I was wrong about this. this is already validated by get_smbios_table(). 

but maybe add a comment here: `/* we assume that get_smbios_table() already validated the header size making some superficial sense */`  or so.